x-pack/plugins/apm/public/context/apm_service/apm_service_context.test.tsx

```diff
@@ -15,23 +15,21 @@ describe('getOrRedirectToTransactionType', () => {
   describe('with transaction type in url', () => {
     it('returns the transaction type in the url ', () => {
       expect(
-        getOrRedirectToTransactionType({
+        getTransactionType({
           transactionTypes: ['worker', 'request', 'custom'],
           transactionType: 'custom',
           agentName: 'nodejs',
-          history,
         })
       ).toBe('custom');
       expect(history.replace).not.toHaveBeenCalled();
```

Comment:
This will fail if you remove the `replace` call from the hook (obviously). It's important that we export a separate function that does the redirect logic so we can more easily test it.